lstm/lstmtrainer.cpp

```diff
@@ -149,7 +149,7 @@ void LSTMTrainer::InitCharSet(const UNICHARSET& unicharset,
 // ctc_mode controls how the truth text is mapped to the network targets.
 // Note: Call before InitNetwork!
 void LSTMTrainer::InitCharSet(const UNICHARSET& unicharset,
-                              const UnicharCompress recoder) {
+                              const UnicharCompress& recoder) {
```

Comment:
`UnicharCompress` can be `NULL`.

In the same file you have this:
`LSTMTrainer::EncodeString(const STRING& str, const UNICHARSET& unicharset, ...`
and
`if (recoder != NULL) {`